### PR TITLE
fix grafana dashboard and clarify dashboard usage more clearly.

### LINKED ISSUE #498: monitoring data node_name not exists ,GPU power usage is not correct in Grafana Dashboard
**Environment**: 
- HAMi version: only know "imageTag=v1.23.10"
- Kubernetes version: 1.23.10
- Others: 2 amd architecture servers  on which Ubuntu20.04.3 LTS were installed, they are k8s nodes
image

**Please provide an in-depth description of the question you have**:
(1)I had installed HAMi successfully, and it works well when runing vGPU task.
from port 31993, I can get monitoring information as followed:
82fe171447300e39046946a6085c277

(2)I deployed dcgm-exporter  by runing “kubectl -n monitoring create -f https://raw.githubusercontent.com/NVIDIA/dcgm-exporter/master/dcgm-exporter.yaml”, and changed the type of svc/dcqm-exporter to NodePort from ClusterIP:
1a6dd7bda42efd8d2986c2601ca3301

(3)I have deployed Prometheus 2.36.1 in binary mode and made the following configurations:
fef412db1fec2d4a750ac2123fb73ed
Targets page from Promethus proves prometheus has already collected monitoring data:
92ec28a5d9c38ed773fb4819024bb5c

(4)I deployed Grafana v8.5.5 in k8s-1.23.10 cluster, A data source named ALL in Grafana was created:
f507d8e4152ffbb4dfcc444caef9792

(5)I imported a dashboard(https://github.com/Project-HAMi/HAMi/blob/master/docs/gpu-dashboard.json), but some of the data presented was inaccurate or was not existed.
for example, "nodename" in the upper left corner has no data, and the value of "GPU power usage" is not inaccurate(My GPU is a NVIDIA A10, whose GPU power usage is 150W)
image

**What do you think about this question?**:
(1)A friend name "凤" from Internet share me this dashboard "https://grafana.com/grafana/dashboards/21833-hami-vgpu-dashboard/", but the problems mentioned above still exist. 
(2)I founded ${node_name} was used in the two dashboard mentioned above, but ${node_name} is NULL. I don't know what's going wrong, please help

### PR DESCRIPTION
**Pull request by jiangsanyin**
Signed-off-by: jiangsanyin <1327212357@qq.com>

**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
fix grafana dashboard and clarify dashboard usage more clearly. Thanks "fangfenghuang (https://github.com/fangfenghuang)" for your help 

**Which issue(s) this PR fixes**:
Fixes #498 #468 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

### CODE REVIEW

File: docs/dashboard.md [hunk 1]
```diff
@@ -1,54 +1,207 @@
...
+metadata:
+  name: gpu-pod-01
+spec:
+  restartPolicy: Never
+  containers:
+  - name: cuda-container
+    image: nvcr.io/nvidia/k8s/cuda-sample:vectoradd-cuda10.2
+    resources:
+      limits:
+        nvidia.com/vgpu: 2 # requesting 2 vGPUs
```
Comment (yxl) [inline docs/dashboard.md:223]: Should "nvidia.com/vgpu" be "nvidia.com/gpu"?

Comment (jiangsanyin) [reply docs/dashboard.md:194]: I forgot to explain it, it depends on our own case. 
In order to distinguish from “nvidia.com/gpu” in nvidia-device-plugin, I used resourceName parameter and setted it's value to "nvidia.com/vgpu", such as: helm install hami hami-charts/hami --set resourceName=nvidia.com/vgpu  --set scheduler.kubeScheduler.imageTag=v1.23.10 -n kube-system

Comment (jiangsanyin) [reply docs/dashboard.md:223]: > Should "nvidia.com/vgpu" be "nvidia.com/gpu"?

I forgot to explain it, it depends on our own case.
In order to distinguish from “nvidia.com/gpu” in nvidia-device-plugin, I used resourceName parameter and setted it's value to "nvidia.com/vgpu", such as: helm install hami hami-charts/hami --set resourceName=nvidia.com/vgpu --set scheduler.kubeScheduler.imageTag=v1.23.10 -n kube-system

**Comment by wawa0210** [comment]
@fangfenghuang  Can you help review this pr?

File: docs/dashboard.md [hunk 2]
```diff
@@ -1,54 +1,207 @@
...
+#apply the file gpu-pod.yaml
+root@controller01:~# kubectl apply -f gpu-pod.yaml
+root@controller01:~# kubectl get pods -o wide
+NAME           READY   STATUS        RESTARTS   AGE   IP               NODE           NOMINATED NODE   READINESS GATES
+gpu-pod-01     0/1     Completed     0          52s   10.233.81.70     controller01   <none>           <none>
 ```
+
+​	You can see the monitoring details in the dashboard. The contents are as follows:
+
+![image-20241003215400685](https://s2.loli.net/2024/10/03/RFJuthzAGYw5UHk.png)
```
Comment (fangfenghuang) [inline docs/dashboard.md]: It is best to place the referenced images in the ../imgs/

**Review by fangfenghuang — COMMENTED** [review]
fix some http url

**Review by fangfenghuang — APPROVED** [review]

File: docs/dashboard.md (hunk 2 above)
Comment (jiangsanyin) [reply docs/dashboard.md]: ok, changes has been made.

**Comment by Nimbus318** [comment]
@jiangsanyin 
I have followed the installation instructions as described in the documentation, but encountered a minor issue, which I also mentioned previously in Issue #498. By default, the `dcgm-exporter` only includes the `Hostname` label. To match the current Grafana dashboard configuration, it's necessary to add a `node_name` relabeling configuration when installing `dcgm-exporter`

https://github.com/NVIDIA/dcgm-exporter/blob/b97b7633e3f39f7a537bd77561cc0ec0c2dca3f5/deployment/values.yaml#L117C3-L117C18

This relabeling should be consistent with the configurations for `hami-device-plugin-svc-monitor` and `hami-scheduler-svc-monitor`

It would be helpful to include this information in the documentation, as users unfamiliar with the Prometheus stack may struggle to configure everything correctly on the first attempt

**Comment by jiangsanyin** [comment]
> @jiangsanyin I have followed the installation instructions as described in the documentation, but encountered a minor issue, which I also mentioned previously in Issue #498. By default, the `dcgm-exporter` only includes the `Hostname` label. To match the current Grafana dashboard configuration, it's necessary to add a `node_name` relabeling configuration when installing `dcgm-exporter`
> 
> https://github.com/NVIDIA/dcgm-exporter/blob/b97b7633e3f39f7a537bd77561cc0ec0c2dca3f5/deployment/values.yaml#L117C3-L117C18
> 
> This relabeling should be consistent with the configurations for `hami-device-plugin-svc-monitor` and `hami-scheduler-svc-monitor`
> 
> It would be helpful to include this information in the documentation, as users unfamiliar with the Prometheus stack may struggle to configure everything correctly on the first attempt


Have you created and applied the ServiceMonitor as depicted in dashboard.md or dashboard_cn.md？node_name is added after this is done.
#Create the file hami-device-plugin-svc-monitor.yaml
root@controller01:~# touch hami-device-plugin-svc-monitor.yaml
#The content of the file hami-device-plugin-svc-monitor.yaml
root@controller01:~# cat hami-device-plugin-svc-monitor.yaml
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  name: hami-device-plugin-svc-monitor
  namespace: kube-system
spec:
  selector:
    matchLabels:
      app.kubernetes.io/component: hami-device-plugin
  namespaceSelector:
    matchNames:
      - "kube-system"
  endpoints:
  - path: /metrics
    port: monitorport
    interval: "15s"
    honorLabels: false
    relabelings:
    - sourceLabels: [__meta_kubernetes_endpoints_name]
      regex: hami-.*
      replacement: $1
      action: keep
    - sourceLabels: [__meta_kubernetes_pod_node_name]
      regex: (.*)
      targetLabel: node_name
      replacement: ${1}
      action: replace
    - sourceLabels: [__meta_kubernetes_pod_host_ip]
      regex: (.*)
      targetLabel: ip
      replacement: $1
      action: replace

#apply the file hami-device-plugin-svc-monitor.yaml
root@controller01:~# kubectl apply -f hami-device-plugin-svc-monitor.yaml

**Comment by Nimbus318** [comment]
@jiangsanyin 
Both are correct. What I meant is that you might have forgotten to include the explanation for the relabel configuration of `dcgm-exporter`. By default, `dcgm-exporter` only includes the `Hostname` label

It’s important to document this configuration to ensure it aligns with the relabeling setup for `hami-device-plugin-svc-monitor`. Without this explanation, users may miss adding the necessary `node_name` relabeling when setting up `dcgm-exporter`

**Comment by jiangsanyin** [comment]
> @jiangsanyin Both are correct. What I meant is that you might have forgotten to include the explanation for the relabel configuration of `dcgm-exporter`. By default, `dcgm-exporter` only includes the `Hostname` label
> 
> It’s important to document this configuration to ensure it aligns with the relabeling setup for `hami-device-plugin-svc-monitor`. Without this explanation, users may miss adding the necessary `node_name` relabeling when setting up `dcgm-exporter`

Ok, thanks to your review. Certain relabelings configurations in serviceMonitor in file deployment/values.yaml of dcgm-exporter has been added, please check!
image

**Pull request merged**